modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb

```diff
@@ -89,8 +89,22 @@ def get_attachment_ids_and_form(parsed_form_data)
       end
 
       def supporting_document_ids(parsed_form_data)
-        parsed_form_data['supporting_docs']&.pluck('attachment_id')&.compact.presence ||
-          parsed_form_data['supporting_docs']&.pluck('claim_id')&.compact.presence || []
+        cached_uploads = []
+        parsed_form_data['supporting_docs']&.each do |d|
+          # Get the database record that corresponds to this file upload:
+          record = PersistentAttachments::MilitaryRecords.find_by(guid: d['confirmation_code'])
```

Comment:
OOC what would happen if this query failed? Any possibility of a silent failure? I don't want to hold this PR up but it may be worth a follow-on to investigate and make it more defensive if necessary.

Reply:
If the query returns `nil` on l95, we'll get an error when we try to pull the `created_at` and `file.id` on lines 98 & 99. This will bubble back up to the user as a 500, so they'll get an error screen and the submission won't go through. 

I can definitely make a follow up to probe this a little more - thanks for looking this over!

Reply:
thanks @michaelclement! maybe the follow-up could be to make a quick couple of request specs that confirm an error is returned when the query returns `nil` or throws.

Reply:
Definitely a good idea - I'm going to put a couple more tests into this PR while it's top of mind. Thanks!

Reply:
@acrollet Tests added in [this commit](https://github.com/department-of-veterans-affairs/vets-api/pull/19647/commits/fe2da6a3b8babdeeb4e3cccfab710ddc8931f49a) - please let me know if there's anything else you think I should take a look at!